### PR TITLE
Remove JVM settings from docker compose

### LINKED ISSUE #580: Kafdrop 4.0.0 Docker image doesn't start
Hello!

Today we restarted one of our Docker Swarm system and the Kafdrop service refused to start up properly. The reason was quickly revealed: the :latest image we used in our compose file was updated so it contains Kafdrop version 4.0.0 now. Here is the service log: 
```
:::    :::     :::     :::::::::: :::::::::  :::::::::   ::::::::  :::::::::           :::    
:+:   :+:    :+: :+:   :+:        :+:    :+: :+:    :+: :+:    :+: :+:    :+:         :+:     
+:+  +:+    +:+   +:+  +:+        +:+    +:+ +:+    +:+ +:+    +:+ +:+    +:+        +:+ +:+  
+#++:++    +#++:++#++: :#::+::#   +#+    +:+ +#++:++#:  +#+    +:+ +#++:++#+        +#+  +:+  
+#+  +#+   +#+     +#+ +#+        +#+    +#+ +#+    +#+ +#+    +#+ +#+             +#+#+#+#+#+
#+#   #+#  #+#     #+# #+#        #+#    #+# #+#    #+# #+#    #+# #+#                   #+#  
###    ### ###     ### ###        #########  ###    ###  ########  ###                   ###  

2023-10-12 10:23:36.222 [32m INFO[m [35m${sys:PID}[m [        main] [36mk.Kafdrop$EnvironmentSetupListener      [m : Initializing JAAS config
2023-10-12 10:23:36.233 [32m INFO[m [35m${sys:PID}[m [        main] [36mk.Kafdrop$EnvironmentSetupListener      [m : Env: null
2023-10-12 10:23:36.370 [32m INFO[m [35m1[m [kground-preinit] [36mo.h.v.i.u.Version                       [m : HV000001: Hibernate Validator 8.0.1.Final
2023-10-12 10:23:36.431 [32m INFO[m [35m1[m [           main] [36mo.s.b.StartupInfoLogger                 [m : Starting Kafdrop v4.0.0 using Java 17.0.8.1 with PID 1 (/kafdrop-4.0.0/kafdrop-4.0.0.jar started by root in /)
2023-10-12 10:23:36.437 [32m INFO[m [35m1[m [           main] [36mo.s.b.SpringApplication                 [m : No active profile set, falling back to 1 default profile: "default"
2023-10-12 10:23:38.393 [31mERROR[m [35m1[m [           main] [36mo.s.b.SpringApplication                 [m : Application run failed

java.lang.StackOverflowError
	at java.base/jdk.internal.ref.PhantomCleanable.<init>(PhantomCleanable.java:66)
	at java.base/jdk.internal.ref.CleanerImpl$PhantomCleanableRef.<init>(CleanerImpl.java:164)
	at java.base/java.lang.ref.Cleaner.register(Cleaner.java:220)
	at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.<init>(ZipFile.java:435)
	at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.<init>(ZipFile.java:428)
	at java.base/java.util.zip.ZipFile.getInputStream(ZipFile.java:395)
	at java.base/java.util.jar.JarFile.getInputStream(JarFile.java:844)
	at java.base/jdk.internal.loader.URLClassPath$JarLoader$2.getInputStream(URLClassPath.java:879)
        ....
```
and here a bit more detailed exception:
```
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:95)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class kotlin.reflect.jvm.internal.impl.builtins.jvm.JavaToKotlinClassMap
	at kotlin.reflect.jvm.internal.RuntimeTypeMapper.mapJvmClassToKotlinClassId(RuntimeTypeMapper.kt:274)
	at kotlin.reflect.jvm.internal.KClassImpl.getClassId(KClassImpl.kt:186)
	at kotlin.reflect.jvm.internal.KClassImpl.access$getClassId(KClassImpl.kt:44)
```

At the moment we reverted image to the version 3.31.0, however would be really good to know: shall we stick to this old version for a long period of time?

Thank you in advance!

### PR DESCRIPTION
**Pull request by mehdihasan**
* Remove JVM settings from the docker compose
* Update Readme to clarify usage of JVM settings

fixes #580

### CODE REVIEW

**Comment by Bert-R** [comment]
The change is good, in the sense that it reportedly solves the user issue.
However... I wonder whether we should go this way or take one more step: totally remove the `-Xss` option. Do we really have reason to 'make it normal' to tune down the stack memory?

I checked the default on my system through this command: `java -XX:+PrintFlagsFinal -version | grep ThreadStackSize`:
```
     intx CompilerThreadStackSize                  = 1024                                   {pd product} {default}
     intx ThreadStackSize                          = 1024                                   {pd product} {default}
     intx VMThreadStackSize                        = 1024                                   {pd product} {default}
openjdk version "17.0.9" 2023-10-17
OpenJDK Runtime Environment Temurin-17.0.9+9 (build 17.0.9+9)
OpenJDK 64-Bit Server VM Temurin-17.0.9+9 (build 17.0.9+9, mixed mode, sharing)
```

I said 'make it normal'. I can imagine that users in some specific circumstances want to tune down the stack memory, but in normal situations, I would favor leaving it to the VM, to prevent issues of the type we just saw.

@mehdihasan @davideicardi Makes sense?

**Comment by mehdihasan** [comment]
I agree. 

It has come to my attention and you may have also encountered the following message:

`OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.`

I would like to propose two potential courses of action:

**Proposal 1**: I recommend considering the removal of the 'noverify' option.

**Proposal 2**: Additionally, I would like to suggest the evaluation of removing all JVM tuning options, as they primarily offer optimization for individual use cases. This approach would empower users to apply their own preferred options or limitations based on their specific requirements.

Your thoughts @Bert-R @davideicardi ?

**Comment by Bert-R** [comment]
Indeed. Let's move that environment variable to the README and shortly describe there how they can tweak the JVM settings.

Can you change your PR that way? Otherwise, I can make a PR on Tuesday.

**Comment by mehdihasan** [comment]
How about now? I tried to be as precise as possible in the README.

**Comment by davideicardi** [comment]
I agree to use the "default" settings whenever it is possible. The less customization of the JVM we have, the better it is for everyone.

Eventually, if we find some options to be useful in certain scenarios, we can add a specific documentation section explain when you should considering adding a specific paramenters.

@mehdihasan Just try to rename the PR title to something more specific. What about: "Remove default JVM settings from docker compose"?

File: README.md [hunk 1]
```diff
@@ -95,16 +95,22 @@ Launch container in background:
 ```sh
 docker run -d --rm -p 9000:9000 \
     -e KAFKA_BROKERCONNECT=<host:port,host:port> \
-    -e JVM_OPTS="-Xms32M -Xmx64M" \
     -e SERVER_SERVLET_CONTEXTPATH="/" \
     obsidiandynamics/kafdrop
 ```
 
+Lunch container with some specific JVM options:
```
Comment (Bert-R) [inline README.md]: Is this a lunch box? :)
```suggestion
Launch container with some specific JVM options:
```

**Review by Bert-R — APPROVED** [review]

**Comment by Bert-R** [comment]
Thanks!

**Pull request merged**